compiler/sem/semexprs.nim

```diff
@@ -1892,48 +1892,44 @@ proc propertyWriteAccess(c: PContext, n, a: PNode): PNode =
     #fixAbstractType(c, result)
     #analyseIfAddressTakenInCall(c, result)
 
-proc takeImplicitAddr(c: PContext, n: PNode; isLent: bool): PNode =
+proc takeImplicitAddr(c: PContext, formal: PType, n: PNode): PNode =
   ## See RFC #7373, calls returning 'var T' are assumed to
   ## return a view into the first argument (if there is one):
   addInNimDebugUtils(c.config, "takeImplicitAddr", n, result)
 
   let root = exprRoot(n)
   if root != nil and root.owner == c.p.owner:
     if root.kind in {skLet, skVar, skTemp} and sfGlobal notin root.flags:
-      localReport(c.config, n.info, reportSym(
+      return newError(c.config, n, reportSym(
         rsemLocalEscapesStackFrame, root, ast = n))
 
     elif root.kind == skParam and root.position != 0:
-      localReport(c.config, n.info, reportSym(
+      return newError(c.config, n, reportSym(
         rsemImplicitAddrIsNotFirstParam, root, ast = n))
 
-  case n.kind
-  of nkHiddenAddr, nkAddr: return n
-  of nkDerefExpr: return n[0]
-  of nkBracketExpr:
-    if n.len == 1: return n[0]
-  of nkHiddenDeref:
-    # issue #13848
-    # `proc fun(a: var int): var int = a`
-    discard
-  else: discard
-  
-  case isAssignable(c, n, isLent)
+  # XXX: what about expressions where ``root == nil`` (e.g. the root is a
```

Comment:
Its my personal preference but I like to use words such as QUESTION, TODO, NOTE, BUG, HACK and similar in the code as (1) a lot of tools can differentiate between these types, (2) it is easier to search and (3) you can see whether it is a QUESTION or a TODO without drilling down into the text, pulling out extra code context and so on.

Reply:
The issue with said words is it's also really painful to search/find them across the codebase. Additionally, they can be a combination of said things.

Reply:
and the issue with `xxx` is it's also really painful to search/find a specific ones out of them in the codebase because they are all uniform. 

I haven't thought about that before but `XXX NOTE` is also a thing and considering the comment's body is likely to dwarf any admonition in size it is not such a big hindrance